src/sage/combinat/integer_vectors_mod_permgroup.py

```diff
@@ -775,6 +785,79 @@ def __iter__(self):
             else:
                 return SF.elements_of_depth_iterator(self._sum)
 
+    def cardinality(self):
+        r"""Return the number of integer vectors in the set.
+
+        The number is computed using the cycle index theorem, which is
+        faster than listing the vectors.
+
+        EXAMPLES::
+
+            sage: G = PermutationGroup([], domain=[1,2,3])
+            sage: IntegerVectorsModPermutationGroup(G, 5).cardinality()
+            21
+
+            sage: G = PermutationGroup([(1,2)])
+            sage: IntegerVectorsModPermutationGroup(G, 1000).cardinality()
+            501
+
+            sage: G = PermutationGroup([(1,2,3)])
+            sage: I = IntegerVectorsModPermutationGroup(G, sum=10, max_part=5)
+            sage: I.cardinality()
+            7
+
+            sage: G = SymmetricGroup(10)
+            sage: I = IntegerVectorsModPermutationGroup(G, max_part=1)
+            sage: I.cardinality()
+            11
+        """
```

Comment:
There are a few more tests (in a `TESTS::` section) that would be good:
```
Check that :issue:`36681` is fixed::
sage: P=PermutationGroup([],domain=[])                                          
sage: V=IntegerVectorsModPermutationGroup(P, sum=1)                             
sage: V.cardinality()   
0
```
Add a group which is not the full symmetric group and not a cyclic group, or maybe better, all permutation groups of small order (can we generate them?  I think you mentioned that?)

Reply:
I am +1000 on this contribution. I'll provide few tests for "fancy groups", if you don't mind.